inductiva/fluids/post_processing/openfoam.py

```diff
@@ -0,0 +1,372 @@
+                        source_radius: float = 0.7,
+                        save_path: inductiva.types.Path = None):
+        """Get streamlines through the fluid/domain in the WindTunnel.
+        
+        The streamlines are obtained by seeding a set of points
+        at the inlet of the WindTunnel.
+
+        Args:
+            simulation_time: Time value to obtain simulation mesh.
+            max_time: Time used for integration of the streamlines.
+                Not related with simulation time.
+            n_points: Number of points to seed.
+            initial_step_length: Initial step length for the streamlines.
+            source_radius: Radius of the source of the streamlines.
+            save_path: Path to save the streamlines. 
+                Types of files permitted: .vtk, .ply, .stl
+        """
+
+        domain_mesh, object_mesh = self.get_mesh_at_time(simulation_time)
+
+        inlet_position = (domain_mesh.bounds[0], 0, 1)
+
+        streamlines_mesh = domain_mesh.streamlines(
+            max_time=max_time,
+            n_points=n_points,
+            initial_step_length=initial_step_length,
+            source_radius=source_radius,
+            source_center=inlet_position)
+
+        if save_path is not None:
+            save_path = inductiva.utils.files.resolve_path(save_path)
+            streamlines_mesh.save(save_path)
+
+        return Streamlines(streamlines_mesh, object_mesh)
+
+    def get_flow_slice(self,
+                       simulation_time: float = 50,
+                       plane: typing.Literal["xy", "xz", "yz"] = "xz",
+                       origin: tuple = (0, 0, 0),
+                       save_path: inductiva.types.Path = None):
+        """Get flow properties in a slice of the domain in WindTunnel.
+        
+        Args:
+            simulation_time: Time value to obtain simulation mesh.
+            plane: Orientation of the plane to slice the domain.
+            origin: Origin of the plane.
+            save_path: Path to save the flow slice. 
+                Types of files permitted: .vtk, .ply, .stl
+        """
+
+        domain_mesh, object_mesh = self.get_mesh_at_time(simulation_time)
+
+        if plane == "xy":
+            normal = (0, 0, 1)
+        elif plane == "yz":
+            normal = (1, 0, 0)
+        elif plane == "xz":
+            normal = (0, 1, 0)
+        else:
+            raise ValueError("Invalid view.")
+
+        flow_slice = domain_mesh.slice(normal=normal, origin=origin)
+
+        if save_path is not None:
+            save_path = inductiva.utils.files.resolve_path(save_path)
+            flow_slice.save(save_path)
+
+        return FlowSlice(flow_slice, object_mesh)
+
+
+@dataclass
+class OpenFOAMPhysicalField(enum.Enum):
+    """Defines the notation used for physical field in OpenFOAM."""
+    PRESSURE = "p"
+    VELOCITY = "U"
+
+
+class MeshData:
+    """MeshData class that allows for mesh data manipulation and render.
```

Comment:
Yes it is, and this will be eventually moved to the more general "CADObject" class where can visualize fields over a general object and have immediately there render capabilities. This was initially the motive for this function.

I moved it here, because it seems more clear before I move it there (still today).